src/todo/todo.lua

```diff
@@ -145,38 +152,45 @@ function todo.import_tasks(dialog)
     todo.update_task_table()
 end
 
-function todo.persist(element)
+function todo.persist(element, player)
     local frame = element.parent.parent
 
     local task_spec, should_add_to_top = todo.get_task_from_add_frame(frame)
-    local task = todo.create_task(task_spec.task, task_spec.assignee)
-
+    local task = todo.create_task(task_spec, player.name)
+    -- Set the creator as last updator too
+    task.updated_by = task.created_by
+    todo.log("Saving task: " .. serpent.block(task))
     todo.save_task(task, should_add_to_top)
 
     todo.log(serpent.block(global.todo))
     frame.destroy()
 end
 
-function todo.update(element, index)
+function todo.update(element, index, player)
     local frame = element.parent.parent
     local task, _ = todo.get_task_from_add_frame(frame)
 
     local original = todo.get_task_by_id(index)
 
+    original.title = task.title
     original.task = task.task
     if (task.assignee) then
         original.assignee = task.assignee
     else
         original.assignee = nil
     end
+    -- Set the last updater
+    original.updated_by = player.name
+    todo.log("Current player is: " .. original.updated_by)
 
     frame.destroy()
 end
 
 function todo.get_task_from_add_frame(frame)
-    local taskText = frame.todo_add_task_table.children[2].text
+    local taskText = frame.todo_add_task_table["todo_new_task_textbox"].text
+    local taskTitle = frame.todo_add_task_table["todo_new_task_title"].text
 
-    local assignees = frame.todo_add_task_table.children[4]
+    local assignees = frame.todo_add_task_table["todo_add_assignee_drop_down"]
```

Comment:
nice!

Reply:
I think we'd better avoid the children[id] everywhere. If we add some things before id 4 this won't work anymore.